File: solutions/csharp/rna-transcription/RnaTranscription.cs
using System;
using System.Text;
using System.Collections.Generic;

public static class RnaTranscription
{
    private static readonly Dictionary<char, char> dnaToRna = new Dictionary<char, char>
    {
        {'G', 'C'},
        {'C', 'G'},
        {'T', 'A'},
        {'A', 'U'},
    };
    public static string ToRna(string dna)
    {
        StringBuilder sb = new StringBuilder(dna.Length);
        foreach (var nucleotide in dna)
        {
            sb.Append(dnaToRna[nucleotide]);
        }
        return sb.ToString();
    }
}

File: solutions/fortran/rna-transcription/rna_transcription.f90
module rna_transcription
   implicit none
contains

   pure function to_rna(dna)
      character(*), intent(in) :: dna
      character(len(dna)) :: to_rna

      character :: nucleotide
      integer :: i

      do i = 1, len(dna)
         select case (dna(i:i))
          case('G')
            nucleotide = 'C'
          case('C')
            nucleotide = 'G'
          case('T')
            nucleotide = 'A'
          case('A')
            nucleotide = 'U'
          case default
            nucleotide = '?'
         end select

         to_rna(i:i) = nucleotide
      end do

   end function to_rna

end module rna_transcription
